tests/websockets/test_graphql_transport_ws.py

```diff
@@ -789,3 +811,76 @@ async def test_subsciption_cancel_finalization_delay(ws: WebSocketClient):
     end = time.time()
     elapsed = end - start
     assert elapsed < delay
+
+
+async def test_subscription_finializer_called(ws: WebSocketClient):
+    # Test that a subscription is promptly finalized when the client interrupts the
+    # subscription
+    await ws.send_json(
+        SubscribeMessage(
+            id="sub1",
+            payload=SubscribeMessagePayload(query="subscription { longFinalizer(delay: 0.0) }"),
+        ).as_dict()
+    )
+
+    response = await ws.receive_json()
+    assert (
+        response
+        == NextMessage(
+            id="sub1", payload={"data": {"longFinalizer": "hello"}}
+        ).as_dict()
+    )
+
+    # check that the context is live, finalize hasn't been called yet.
+    # finalizer_state is set True to when subscription is running, then back
+    # to False when the subscription is finalized
+    await ws.send_json(
+        SubscribeMessage(
+            id="sub2",
+            payload=SubscribeMessagePayload(query="query { finalizerState }"),
+        ).as_dict()
+    )
+
+    # wait for the sub2 message
+    while True:
+        response = await ws.receive_json()
+        assert response["type"] in ["next", "complete"]
+        if response["type"] != "next" or response["id"] != "sub2":
+            continue
+        assert response["payload"]["data"]["finalizerState"] is True
+        break
+
+    # cancel the subscription
+    await ws.send_json(CompleteMessage(id="sub1").as_dict())
+
+    # wait until context is dead.
+    # We don't know exactly how many packets will arrive or how long it will take.
+    # Need manual timeout because async timeout doesn't work on async integrations
```

Comment:
> Need manual timeout because async timeout doesn't work on async integrations

how come?

Reply:
Because the actual IO calls are actually blocking, and the event loop running the test will stop until the call returns, so no "async" timeouts can be delivered.  And not all of the integrations support a "timeout" parameter to their receive methods.